### PR TITLE
Add Intial Readme

### PR DESCRIPTION
**Pull request by noammirjani**
added a readme file that fits to the current stage and vision of the project. 
- missing a usage section because it will be added in the future after having a base of the

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+2. Install the required dependencies:
+    ```sh
+    poetry install
+    poetry build
+    ```
+
+### Usage
+- will be explained soon :)
+
+### Roadmap
```
Comment (daviddsapir) [inline README.md:42]: This is more like feature/future-features.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+### Features
+
+- **Create, Edit, and Delete Tasks**: Manage your tasks with ease.
+- **Retrieve Tasks by ID**: Fetch any task using its unique ID.
+- **Command-line Interface**: Lightweight and fast interface to manage your to-do lists.
+
+### Installation
+
+#### Prerequisites
```
Comment (daviddsapir) [inline README.md:22]: You can remove the `Prerequisites` and `Setup`.

This is for "us". You need to add only stuff that the "user" needs to know.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+- [Usage](#usage)
+- [Roadmap](#Roadmap)
+- [Contributing](#contributing)
+- [License](#license)
+
+### Features
+
+- **Create, Edit, and Delete Tasks**: Manage your tasks with ease.
+- **Retrieve Tasks by ID**: Fetch any task using its unique ID.
+- **Command-line Interface**: Lightweight and fast interface to manage your to-do lists.
```
Comment (daviddsapir) [inline README.md:18]: We've decided that this is mangment and not sloly a **to-do** application.
We will provide calendar and also reminders (which is somthing like to-do).

**Comment by daviddsapir** [comment]
Read this [article](https://cbea.ms/git-commit/)
It explains how to write a good commit title & message

File: README.md (hunk 1 above)
Comment (noammirjani) [reply README.md:42]: I do not agree on this one.. features title should explain what is the project offer and future features is  with the same meaning. In the meanwhile, the meaning of usage in readme is how to run or how to play with project functionality like user instructions

File: README.md (hunk 3 above)
Comment (noammirjani) [reply README.md:18]: I will add the calendar as a feature

File: README.md (hunk 2 above)
Comment (noammirjani) [reply README.md:22]: I had change the title from 'Prerequisites' to 'dependencies' but I do feel like that is a critical data that the user should be known about

Comment (daviddsapir) [reply README.md:22]: We will provide the user with an installation script, potentially using Homebrew.

If any issues arise during installation, we will promptly inform them.

You’re right—it’s a good idea to also specify that the required Python version is above 3.8, so they are aware of this dependency beforehand.

Saying that poetry is a Prerequisites is not right. It is a Prerequisites only for dev (for us).

File: README.md (hunk 1 above)
Comment (daviddsapir) [reply README.md:42]: OK. then keep it.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+### Installation
+
+#### Prerequisites
+- Python 3.8+
+- Poetry (dependency management)
```
Comment (daviddsapir) [inline README.md:24]: Adding poetry here is redundant for the user prespective.

Comment (noammirjani) [reply README.md:24]: but again for contributors, they can not maintain the program without poetry

Comment (daviddsapir) [reply README.md:24]: You're right 🙂. This can be helpful for contributors. Keep it

Comment (daviddsapir) [reply README.md]: I suggest adding to the `bootstrap` recipe in the `Makefile` those checks - `python version > 3.8` and that `poetry` is avaiable.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+### Roadmap
+- Google Authentication: Secure user authentication via Google.
+- Improved UI: Create a more user-friendly command-line interface.
+- calendar view: Add a calendar view to see tasks by date.
+- (optional) notifications system: Add notifications for upcoming tasks.
+- Tagging System: Add tags to tasks for better organization.
+- Task Sorting: Multiple sorting options, such as by date, tag, or priority.
+- Readme update: Add usage instructions, examples and demo/gif.
+-
```
Comment (daviddsapir) [inline README.md:46]:
```suggestion
```

**Pull request merged**